test/unit/org/apache/cassandra/repair/autorepair/AutoRepairParameterizedTest.java

```diff
@@ -782,4 +783,48 @@ public void testRepairThrowsForIRWithCDCReplay()
             AutoRepair.instance.repair(repairType);
         }
     }
+
+    @Test
+    public void testSoakAfterImmediateRepair()
+    {
+        when(autoRepairState.getRepairRunnable(any(), any(), any(), anyBoolean())).thenReturn(repairRunnable);
+        when(autoRepairState.isSuccess()).thenReturn(true);
+        AutoRepairConfig config = AutoRepairService.instance.getAutoRepairConfig();
+        config.repair_task_min_duration = new DurationSpec.LongSecondsBound("1s");
```

Comment:
The 1s duration is too small to make this test case flaky. Also, we do not need to test every single table.
I would say, set this to "10s" or something, and as soon as the first table hits the sleep call, set it back to "0s"